src/app/core/internal/view/rendering/RenderHelper.ts

```diff
@@ -0,0 +1,120 @@
+import {Matrix2x3}              from "math/Matrix";
+import {SVGDrawing, parseColor} from "svg2canvas";
+import {V, Vector}              from "Vector";
+import {Prim}                   from "../Prim";
+import {Style}                  from "./Style";
+
+
+export class RenderHelper {
+    protected canvas?: HTMLCanvasElement;
+    protected context?: CanvasRenderingContext2D;
+
+    // protected camera: CameraView;
+
+    public constructor() {
+        // this.camera = camera;
+    }
+
+    protected get ctx(): CanvasRenderingContext2D {
+        if (!this.context)
+            throw new Error(`RenderHelper: Uninitialized context! ${this.canvas} | ${this.context}`);
+        return this.context;
+    }
+
+    public setCanvas(canvas?: HTMLCanvasElement) {
+        this.canvas = canvas;
+        this.context = canvas?.getContext("2d") as CanvasRenderingContext2D | undefined;
+    }
+
+    public clear(): void {
+        this.ctx.clearRect(0, 0, this.size.x, this.size.y);
+    }
+
+    public transform(m: Matrix2x3) {
+        this.ctx.transform(
+            m.get(0), m.get(1),
+            m.get(2), m.get(3),
+            m.get(4), m.get(5),
+        );
+    }
+
+    public toWorldSpace(mat: Matrix2x3): void {
+        const inv = mat.inverse();
+        this.transform(inv.withTranslation(inv.pos.add(this.size.scale(0.5))));
+    }
+    // public toScreenSpace(): void {
+    //     this.transform(this.camera.matrix);
+    // }
+
+    public image(img: SVGDrawing, pos: Vector, size: Vector, tint?: string): void {
+        const col = (tint ? parseColor(tint) : undefined);
+
+        // Flip y-axis scale
+        img.draw(this.ctx, pos.x, pos.y, size.x, -size.y, col);
+    }
+
+    public draw(prim: Prim): void {
+        prim.render(this.ctx);
+        // this.save();
+        // this.setStyle(style, alpha);
+
+        // // Begin path and draw the shape
+        // this.ctx.beginPath();
+        // shape.draw(this.ctx);
+
+        // // Only fill or stroke if we have to
+        // if (style.fill())
+        //     this.ctx.fill();
+        // if (style.stroke())
+        //     this.ctx.stroke();
+
+        // this.ctx.closePath();
+        // this.restore();
```

Comment:
Is this commented out code something to be TODOne later or should it be deleted?